src/full/Agda/TypeChecking/Serialise/Instances/Common.hs

```diff
@@ -524,11 +524,21 @@ instance EmbPrj Quantity where
 --   value 2 = return Quantityω
 --   value _ = malformed
 
+instance EmbPrj Cohesion where
+  icod_ Flat       = return 0
```

Comment:
For `Enum, Bounded` types you need not define the serialization manually, just
```haskell
instance EmbPrj Cohesion
```
is sufficient.
I did not eagerly remove the manual enumerations, though, I only use the default instance for new stuff I add.